Index: app/src/main/java/com/example/estgmapper/MainActivity.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/estgmapper/MainActivity.java b/app/src/main/java/com/example/estgmapper/MainActivity.java
--- a/app/src/main/java/com/example/estgmapper/MainActivity.java	
+++ b/app/src/main/java/com/example/estgmapper/MainActivity.java	(date 1696270354323)
@@ -1,15 +1,19 @@
 package com.example.estgmapper;
 
 import android.os.Bundle;
+import android.os.PersistableBundle;
 import android.view.MenuItem;
 import android.view.MotionEvent;
 import android.view.View;
 import android.view.WindowManager;
 import android.widget.Button;
 import android.widget.ImageView;
+import android.widget.TextView;
 import android.widget.ZoomControls;
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.DialogFragment;
 import androidx.navigation.NavController;
 import androidx.navigation.Navigation;
 import androidx.navigation.ui.AppBarConfiguration;
@@ -17,7 +21,9 @@
 import com.example.estgmapper.databinding.ActivityMainBinding;
 import com.google.android.material.bottomnavigation.BottomNavigationView;
 
-public class MainActivity extends AppCompatActivity {
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity implements MultipleChoiceDialogFragment.onMultiChoiceListener{
 
     private ActivityMainBinding binding;
     private ImageView imageView;
@@ -29,6 +35,7 @@
     private float initialFingerSpacing = -1;
     private float originalScaleX, originalScaleY;
     private Button button;
+    private TextView tvSelectedChoices;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
@@ -156,6 +163,20 @@
                 }
             }
         });
+
+        setContentView(R.layout.activity_main); // Make sure this is the correct layout file
+
+        tvSelectedChoices=findViewById(R.id.tvSelectedChoices);
+        Button btnSelectFilter=findViewById(R.id.filters);
+
+        btnSelectFilter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                DialogFragment multiChoiceDialog = new MultipleChoiceDialogFragment();
+                multiChoiceDialog.setCancelable(false);
+                multiChoiceDialog.show(getSupportFragmentManager(), "Multichoice Dialog");
+            }
+        });
     }
 
     private float calculateFingerSpacing(MotionEvent event) {
@@ -163,4 +184,19 @@
         float y = event.getY(0) - event.getY(1);
         return (float) Math.sqrt(x * x + y * y);
     }
+
+    @Override
+    public void onPositiveButtonClicked(String[] list, ArrayList<String> selectedItemList) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Selected Choices = ");
+        for (String str:selectedItemList){
+            stringBuilder.append(str+" ");
+        }
+
+    }
+
+    @Override
+    public void onNegativeButtonClicked() {
+
+    }
 }
